homeassistant/components/imap/coordinator.py

```diff
@@ -135,10 +142,16 @@ async def async_start(self) -> None:
     async def _async_wait_push_loop(self) -> None:
         """Wait for data push from server."""
         while True:
-            await self._async_reconnect_if_needed()
             try:
                 number_of_messages = await self._async_fetch_number_of_messages()
-            except UpdateFailed as ex:
+            except (
+                UpdateFailed,
+                InvalidAuth,
+                InvalidFolder,
+                AioImapException,
+                asyncio.TimeoutError,
+            ) as ex:
+                self.async_set_updated_data(None)
```

Comment:
We catch all possible errors here to ensure the task is not breaking and we ensure the sensor becomes `unknown`.

Reply:
`async_set_update_error` should take care of setting the sensor to `unavailable`

Reply:
If its `InvalidAuth` we need to take care of starting reauth ourselves since its not going to be done by the coordinator

https://github.com/home-assistant/core/blob/81c0382e4b38d6d0045766684c2a0fa6ffb28ad6/homeassistant/helpers/update_coordinator.py#L313